src/mantarray_waveform_analysis/peak_detection.py

```diff
@@ -99,18 +99,24 @@ def peak_detector(
     # TODO Tanner (11/3/20): move this loop to find_twitch_indices
     # Patches error in B6 file for when two valleys are found in a single valley. If this is true left_bases, right_bases, prominences, and raw magnetic sensor data will also be equivalent to their previous value. This if statement indicates that the valley should be disregarded if the interpolated values on left and right intersection points of a horizontal line at the an evaluation height are equivalent. This would mean that the left and right sides of the peak and its neighbor peak align, indicating that it just one peak rather than two.
     # https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.peak_widths.html#scipy.signal.peak_widths
-    for i in range(1, len(valley_indices)):
+
+    i = 1
+    while i < len(valley_indices):
         if left_ips[i] == left_ips[i - 1] and right_ips[i] == right_ips[i - 1]:
             valley_idx = valley_indices[i]
             valley_idx_last = valley_indices[i - 1]
-            if magnetic_signal[valley_idx_last] == magnetic_signal[valley_idx]:
+
+            if magnetic_signal[valley_idx_last] >= magnetic_signal[valley_idx]:
                 valley_indices = np.delete(valley_indices, i)
-                i -= 1
-            elif magnetic_signal[valley_idx] > magnetic_signal[valley_idx_last]:
-                valley_indices = np.delete(valley_indices, i - 1)
-            else:  # pragma: no cover  # (Anna 3/31/21): we don't have a case as of yet in which the first peak is higher than the second however know that it is possible and therefore aren't worried about code coverage in this case.
-                valley_indices = np.delete(valley_indices, i)  # pragma: no cover
-                i -= 1  # pragma: no cover
+                left_ips = np.delete(left_ips, i)
+                right_ips = np.delete(right_ips, i)
+            else:  # pragma: no cover # (Anna 3/31/21): we don't have a case as of yet in which the first peak is higher than the second however know that it is possible and therefore aren't worried about code coverage in this case.
+                valley_indices = np.delete(valley_indices, i - 1)  # pragma: no cover
+                left_ips = np.delete(left_ips, i - 1)  # pragma: no cover
+                right_ips = np.delete(right_ips, i - 1)  # pragma: no cover
+
+            i -= 1
```

Comment:
I just want to make sure I understand this, inside the if statement this will always delete one of the elements of `valley_indices`, so in that case `i` should stay the same value as the current iteration so could this instead have an `else` statement to the `if left_ips[i] == left_ips[i - 1] and right_ips[i] == right_ips[i - 1]:` statement where it increments `i` and remove the `i -= 1` from the if statement?

something like
```python
    while i < len(valley_indices):
        if left_ips[i] == left_ips[i - 1] and right_ips[i] == right_ips[i - 1]:
            valley_idx = valley_indices[i]
            valley_idx_last = valley_indices[i - 1]

            if magnetic_signal[valley_idx_last] >= magnetic_signal[valley_idx]:
                valley_indices = np.delete(valley_indices, i)
                left_ips = np.delete(left_ips, i)
                right_ips = np.delete(right_ips, i)
            else:
                valley_indices = np.delete(valley_indices, i - 1)  # pragma: no cover
                left_ips = np.delete(left_ips, i - 1)  # pragma: no cover
                right_ips = np.delete(right_ips, i - 1)  # pragma: no cover
        else:
            i += 1

Reply:
oh yes that totally makes sense, I'll change that now!